explorer_frontend/src/features/contracts/components/Deploy/DeployTab.tsx

```diff
@@ -22,45 +23,75 @@ import {
 } from "../../models/base";
 import { ShardIdInput } from "./ShardIdInput";
 
-export const DeployTab = () => {
-  const [smartAccount, args, constuctorAbi, pending, shardId] = useUnit([
-    $smartAccount,
-    $deploymentArgs,
-    $constructor,
-    deploySmartContractFx.pending,
-    $shardId,
-  ]);
+type DeployTabProps = {
+  onClose?: () => void;
+};
+
+export const DeployTab = ({ onClose }: DeployTabProps) => {
+  const [smartAccount, args, constuctorAbi, pending, shardId, balance, endpoint, errorMessage] =
+    useUnit([
+      $smartAccount,
+      $deploymentArgs,
+      $constructor,
+      deploySmartContractFx.pending,
+      $shardId,
+      $balance,
+      $endpoint,
+      $deployErrorMessage,
+    ]);
+
+  const isDeployDisabled =
+    pending || !smartAccount || !endpoint || !balance || balance <= 0 || shardId === null;
+
   const [css] = useStyletron();
+  const modalOpened = useRef(false);
+
+  useEffect(() => {
+    modalOpened.current = true;
+  }, []);
+
+  const caption = useMemo(() => {
```

Comment:
useMemo is excessive here (it optimizes nothing but takes resources for an optimization function). please use ternary `errorMessage ? ... ?`